internal/reconciler/metricpipeline/reconciler.go

```diff
@@ -238,8 +238,8 @@ func (r *Reconciler) isReconcilable(ctx context.Context, pipeline *telemetryv1al
 		return false, nil
 	}
 
-	if secretref.ReferencesNonExistentSecret(ctx, r.Client, pipeline) {
-		return false, nil
+	if err := secretref.VerifySecretReference(ctx, r.Client, pipeline); err != nil {
+		return false, nil //nolint:nilerr //secret does not exist, however, it is not an error
```

Comment:
are we sure the secret does not exist? it could be any sort of error thrown by the function

Reply:
fixed